### PR TITLE
Od/arrays.md heb

### PR DESCRIPTION
**Pull request by odeddugma**
This is 1-65 lines translation at the Arrays chapter.

### CODE REVIEW

File: manuscript_heb/10-Arrays.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+console.log(items[1]); //"2"
+```
+
+כאשר מועבר לקונסטרקטור של המערך ערך מספרי כארגומנט יחיד ערך זה יהווה את האורך
+`length`
+של המערך. לעומת זאת, אם מועבר לארגומנט של הקונסטרקטור ערך יחיד שאינו מספרי, ערך זה יהפוך לפריט האחד והיחיד במערך. אם מספר ערכים (מספריים או לא) מועברים לקונסטרקטור כארגומנטים ערכים אלו יהפכו אף הם לפריטים במערך. התנהגות זו הינה מסוכנת ומבולבלת כיוון שלא תמיד נוכל לדעת אלו סוגים של מידע הועברו לקונסטרקור בתור ארגומנטים.
+
+ECMAScript6
+מציגה את המתודה
+`Array.of()`
```
Comment (ronen-e) [inline manuscript_heb/10-Arrays.md:70]: you need to wrap these in `ltr` spans like this:

```html
<span dir="ltr">`Array.of()`</span>
```


see what happens when you view the file on github:
https://github.com/ronen-e/understandinges6/blob/549f2eb3ca773b31ebd19fc9044e358d3b222ee2/manuscript_heb/10-Arrays.md

File: manuscript_heb/10-Arrays.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+<div dir="rtl">
+
+# יכולות מערך משופרות
+
+המערך הינו אובייקט ג'אווה סקריפט במהותו. עם זאת, בעוד אספקטים אחרים של ג'אווה סקריפט התפתחו במהלך הזמן, מערכים נותרו זהים עד ש-
```
Comment (ronen-e) [inline manuscript_heb/10-Arrays.md:5]: תיקונים:
ג׳אווהסקריפט
במקום 
ג׳אווה סקריפט

היבטים
במקום
אספקטים

File: manuscript_heb/10-Arrays.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+ECMAScript5
+הציגה מספר שיטות להפוך אותם לקלים יותר לשימוש.
+ECMAScript6
+המשיכה את מגמת השיפור של המערכים על ידי הוספת פונקציונליות רבה יותר, כמו שיטות יצירה חדשות של מערכים, הפיכת כמה שיטות שימושיות לנוחות יותר לכתיב, מערכים מאופיינים.
```
Comment (ronen-e) [inline manuscript_heb/10-Arrays.md:9]: ECMAScript 5
במקום
ECMAScript5

ECMAScript 6
במקום
ECMAScript6


נוחות יותר לכתיבה
במקום
נוחות יותר לכתיב

File: manuscript_heb/10-Arrays.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+items = new Array(3, "2");
+console.log(items.length); // 2
+console.log(items[0]); // 3
+console.log(items[1]); //"2"
+```
+
+כאשר מועבר לקונסטרקטור של המערך ערך מספרי כארגומנט יחיד ערך זה יהווה את האורך
+`length`
+של המערך. לעומת זאת, אם מועבר לארגומנט של הקונסטרקטור ערך יחיד שאינו מספרי, ערך זה יהפוך לפריט האחד והיחיד במערך. אם מספר ערכים (מספריים או לא) מועברים לקונסטרקטור כארגומנטים ערכים אלו יהפכו אף הם לפריטים במערך. התנהגות זו הינה מסוכנת ומבולבלת כיוון שלא תמיד נוכל לדעת אלו סוגים של מידע הועברו לקונסטרקור בתור ארגומנטים.
+
```
Comment (ronen-e) [inline manuscript_heb/10-Arrays.md:67]: לקונסטרקטור
במקום
לקונסטרקור

מסוכנת ומבלבלת
במקום
מסוכנת ומבולבלת

File: manuscript_heb/10-Arrays.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+console.log(items[0]); // 2
+
+items = Array.of("2");
+console.log(items.length); // 1
+console.log(items[0]); // "2"
+```
+
+בכדי ליצור מערך באמצעות מתודת
+`Array.of()`
+פשוט נעביר לה את הערכים הרצויים במערך. כך למשל בדוגמא הראשונה לעיל נוצר מערך המכיל שני מספרים. המערך שבדוגמא השניה מכיל מספר אחד והמערך האחרון מכיל סטרינג אחד. הדבר דומה לשימוש במערך מילולי ואכן ניתן להשתמש במערך מילולי במקום
```
Comment (ronen-e) [inline manuscript_heb/10-Arrays.md:96]: מחרוזת אחת
במקום
סטרינג אחד

מערך ליטראלי
במקום
מערך מילולי

File: manuscript_heb/10-Arrays.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+`Array.of()`
+ליצירת מערכים מקוריים במרבית המקרים.
+אך אם נרצה להעביר את הקונסטרקטור של המערך לפונקציה אז יתכן שנרצה להתשתמש במתודה
```
Comment (ronen-e) [inline manuscript_heb/10-Arrays.md:99]: להשתמש
במקום
להתשתמש

File: manuscript_heb/10-Arrays.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+`Array.of()`
+בתור הארגומנט הראשון לפונקציה
+`createArray()`
+על מנת ליצור מערך חדש. במקרה זה יהיה זה מסוכן להעביר רק
+`Array`
+ישירות מבלי שנוכל להבטיח שהערך
+`(value)`
+לא יהיה מספר.
+
+- לתשומת לב: המתודה
```
Comment (ronen-e) [inline manuscript_heb/10-Arrays.md:125]: שימו לב
במקום
לתשומת לב

File: manuscript_heb/10-Arrays.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+  `Array.of()`
+  אינה משתמשת במאפיין
+  `Sumbol.species`
```
Comment (ronen-e) [inline manuscript_heb/10-Arrays.md:128]: `Symbol.species`
במקום
`Sumbol.species`

**Review by ronen-e — CHANGES_REQUESTED** [review]

**Comment by ronen-e** [comment]
@odeddugma can you sync with upstream and rebase your changes on top?

File: manuscript_heb/10-Arrays.md (hunk 3 above)
Comment (odeddugma) [reply manuscript_heb/10-Arrays.md:9]: היה צריך לכתוב "לכתיבת". תיקנתי בהתאם.

**Comment by odeddugma** [comment]
Hey, I made the changes you asked locally. I will make last proofreading and then I'll open a new pull request.


You wrote: @odeddugma can you sync with upstream and rebase your changes on top?

I'm not that famillier with github but I guess you mean that pull request, isn't it?

**Comment by ronen-e** [comment]
> Hey, I made the changes you asked locally. I will make last proofreading and then I'll open a new pull request.
> 
> You wrote: @odeddugma can you sync with upstream and rebase your changes on top?
> 
> I'm not that famillier with github but I guess you mean that pull request, isn't it?

@odeddugma  thanks for the changes.

What I mean with rebase is this:
[How do I update a GitHub forked repository?](https://stackoverflow.com/questions/7244321/how-do-i-update-a-github-forked-repository)

Basically, the PR should contain have only your commits. For example this PR should not show commit [43feb1d](https://github.com/ronen-e/understandinges6/pull/24/commits/43feb1df57b656e243d08156f0f7a65aea2ced22)

The commits for this PR should start at your first commit which is https://github.com/ronen-e/understandinges6/pull/24/commits/bf7978aaeeff59b8de0836c968f432c3ceea627e

This could also help with resetting a branch in case you need to do so.
[How do I reset 'master' to 'origin/master'?
](https://superuser.com/questions/273172/how-do-i-reset-master-to-origin-master/273199)

**Pull request closed without merging**